### PR TITLE
mrc-2352 Add chart components to display results data

### PR DESCRIPTION
**Pull request by EmmaLRussell**
This branch adds `Charts` and `Chart` components for displaying results data using the chart metadata. 

`Charts` simply iterates through each chart metadata object and renders a `Chart` component for each. 

`Chart` re-implements the proof of concept as a vue component. 

Some reorgansiation of the metadata was necessary as the input data for all charts now comes from a unified dataset, from which each chart's metadata chooses the relevant selections it needs. 

I've distinguished between "chartData" (the results data) and "layoutData" which is any auxiliary data the chart needs in order to draw in phase lines, make calculations based on population etc. This is really the parameters data - I'm providing a hardcoded params set until we implement dynamic parameters. Maybe "chartData" and "layoutData" aren't the best names for these (since layout data might be used in config too, and is also used to calcuated processed data values) - "primaryData" and "extraData"?

Not yet implemented: making charts collapsible, checkboxes to toggle series visibility.

Massive package diff - Plotly deps mostly I guess :grimacing:

### CODE REVIEW

File: src/app/static/src/views/Home.vue [hunk 1]
```diff
@@ -1,19 +1,28 @@
...
+    <Charts v-if="metadata"
+            :chart-metadata="metadata.charts"
+            :chart-data="results"
+            :layout-data="chartLayoutData"></Charts>
   </div>
 </template>
 
 <script lang="ts">
 import { defineComponent } from "vue";
-import HelloWorld from "@/components/HelloWorld.vue";
```
Comment (mwoodbri) [inline src/app/static/src/views/Home.vue:10]: Probably time for a ticket to expunge this component and all the associated `increment`, `count` etc.

**Review by mwoodbri — APPROVED** [review]
This is really great - feels like a real milestone to see some charts, even if data is hard-coded!

Not sure what happened to Rt line in translation from HTML version, but hardly an issue given that we don't have the real data yet.

Feels like a bit of a shame that the charts have to magically know the name of their `data` field... but I guess that's a consequence of wanting to pass all the data to all of the charts? It makes the development a hypothetical tool to allow graph development/visualisation slightly harder. Alternative would be for the higher-level component to pull out the relevant part of the data from the bigger tree and pass it down to each component. Would that even work?

File: src/app/static/src/views/Home.vue (hunk 1 above)
Comment (EmmaLRussell) [reply src/app/static/src/views/Home.vue:10]: https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-2372

**Comment by EmmaLRussell** [comment]
> Feels like a bit of a shame that the charts have to magically know the name of their data field... but I guess that's a consequence of wanting to pass all the data to all of the charts? 

Yes, and when I started off it was also because the layout-related data (the param values) were also part of the same dataset - although now that we're separated those to input them, as you say we could potentially have something defined in the chart metadata (e.g. `dataset: "time_series"`), and then the Charts component could pull that out of chartData and pass it down to Chart (or Chart itself could handle that part) so the chart metadata could just know it as 'data'...

I think I want to press on with getting more basic functionality in for now, but I'll make a ticket to address this.

**Pull request merged**